clients/elrond/client_test.go

```diff
@@ -565,3 +591,112 @@ func TestClient_Close(t *testing.T) {
 	assert.Nil(t, err)
 	assert.True(t, closeCalled)
 }
+
+func TestClient_CheckClientAvailability(t *testing.T) {
+	t.Parallel()
+
+	currentNonce := uint64(0)
+	incrementor := uint64(1)
+	args := createMockClientArgs()
+	statusHandler := testsCommon.NewStatusHandlerMock("test")
+	args.StatusHandler = statusHandler
+	expectedErr := errors.New("expected error")
+	args.Proxy = &interactors.ElrondProxyStub{
+		GetShardOfAddressCalled: func(ctx context.Context, bech32Address string) (uint32, error) {
+			return 0, nil
+		},
+		GetNetworkStatusCalled: func(ctx context.Context, shardID uint32) (*data.NetworkStatus, error) {
+			currentNonce += incrementor
+			return &data.NetworkStatus{
+				Nonce: currentNonce,
+			}, nil
+		},
+	}
+	c, _ := NewClient(args)
+	t.Run("different current nonce should update - 10 times", func(t *testing.T) {
```

Comment:
use exactly AllowedDelta value

Reply:
this is a repetitive test. Could have been 100 as well